File: android/app/src/main/java/com/gamespace/app/utils/DeviceDetector.java
package com.gamespace.app.utils;

import android.os.Build;
import com.gamespace.app.core.DeviceSpecModel;

import java.util.HashMap;
import java.util.Map;

public class DeviceDetector {

    public enum ChipsetVendor {
        QUALCOMM,
        MEDIATEK,
        EXYNOS,
        UNISOC,
        TENSOR,
        KIRIN,
        GENERIC
    }

    public static DeviceSpecModel getDeviceSpecModel() {
        ChipsetVendor vendor = detectChipsetVendor();
        String chipsetName = getChipsetName();
        return new DeviceSpecModel(
                Build.MANUFACTURER,
                Build.MODEL,
                Build.BOARD,
                Build.HARDWARE,
                Build.VERSION.RELEASE,
                Build.VERSION.SDK_INT,
                chipsetName,
                vendor
        );
    }

    public static Map<String, String> getDeviceSpecs() {
        Map<String, String> specs = new HashMap<>();
        specs.put("model", Build.MODEL);
        specs.put("manufacturer", Build.MANUFACTURER);
        specs.put("board", Build.BOARD);
        specs.put("hardware", Build.HARDWARE);
        specs.put("android_version", Build.VERSION.RELEASE);
        specs.put("sdk_int", String.valueOf(Build.VERSION.SDK_INT));

        ChipsetVendor vendor = detectChipsetVendor();
        specs.put("chipset_vendor", vendor.name());
        specs.put("chipset_name", getChipsetName());

        return specs;
    }

    private static String getChipsetName() {
        String hardware = Build.HARDWARE.toLowerCase();
        String board = Build.BOARD.toLowerCase();

        if (hardware.contains("qcom") || board.contains("sm") || board.contains("sdm")) {
            return "Qualcomm Snapdragon";
        } else if (hardware.contains("mt") || board.contains("mt")) {
            return "MediaTek Dimensity/Helio";
        } else if (hardware.contains("exynos") || board.contains("universal")) {
            return "Samsung Exynos";
        } else if (hardware.contains("ums") || board.contains("sp")) {
            return "Unisoc Tiger";
        } else if (hardware.contains("gs") || board.contains("slider")) {
            return "Google Tensor";
        } else if (hardware.contains("kirin") || board.contains("hi")) {
            return "HiSilicon Kirin";
        } else {
            return Build.HARDWARE;
        }
    }

    public static ChipsetVendor detectChipsetVendor() {
        String hardware = Build.HARDWARE.toLowerCase();
        String board = Build.BOARD.toLowerCase();

        if (hardware.contains("qcom") || board.contains("sm") || board.contains("sdm")) {
            return ChipsetVendor.QUALCOMM;
        } else if (hardware.contains("mt") || board.contains("mt")) {
            return ChipsetVendor.MEDIATEK;
        } else if (hardware.contains("exynos") || board.contains("universal")) {
            return ChipsetVendor.EXYNOS;
        } else if (hardware.contains("ums") || board.contains("sp")) {
            return ChipsetVendor.UNISOC;
        } else if (hardware.contains("gs") || board.contains("slider")) {
            return ChipsetVendor.TENSOR;
        } else if (hardware.contains("kirin") || board.contains("hi")) {
            return ChipsetVendor.KIRIN;
        } else {
            return ChipsetVendor.GENERIC;
        }
    }
}

File: android/app/src/main/kotlin/com/gamespace/app/utils/DeviceDetector.kt
package com.gamespace.app.utils

import android.os.Build
import java.io.RandomAccessFile

object DeviceDetector {

    fun getDeviceInfoMap(): Map<String, Any> {
        val hardware = Build.HARDWARE ?: "Unknown"
        val board = Build.BOARD ?: "Unknown"
        val manufacturer = Build.MANUFACTURER ?: "Unknown"
        val model = Build.MODEL ?: "Unknown"
        val chipset = detectChipset(hardware, board)
        val cores = Runtime.getRuntime().availableProcessors()
        val totalRamMb = getTotalMemoryMb()

        return mapOf(
            "manufacturer" to manufacturer,
            "model" to model,
            "hardware" to hardware,
            "board" to board,
            "chipset" to chipset,
            "cpuCores" to cores,
            "totalRamMb" to totalRamMb,
            "androidVersion" to Build.VERSION.RELEASE,
            "sdkInt" to Build.VERSION.SDK_INT
        )
    }

    private fun detectChipset(hardware: String, board: String): String {
        val hwLower = hardware.lowercase()
        val boardLower = board.lowercase()

        return when {
            hwLower.contains("mt") || boardLower.contains("mt") -> "MediaTek (Helio / Dimensity)"
            hwLower.contains("qcom") || hwLower.contains("snapdragon") || boardLower.contains("msm") || boardLower.contains("sdm") -> "Qualcomm Snapdragon"
            hwLower.contains("ums") || hwLower.contains("sp98") || hwLower.contains("unisoc") || boardLower.contains("sc98") -> "Unisoc (Tiger Series)"
            hwLower.contains("exynos") || boardLower.contains("universal") || boardLower.contains("s5e") -> "Samsung Exynos"
            hwLower.contains("gs101") || hwLower.contains("gs201") || hwLower.contains("zuma") || boardLower.contains("tensor") -> "Google Tensor"
            hwLower.contains("kirin") || boardLower.contains("hi36") || boardLower.contains("hi62") -> "HiSilicon Kirin"
            else -> "$hardware ($board)"
        }
    }

    private fun getTotalMemoryMb(): Long {
        return try {
            val reader = RandomAccessFile("/proc/meminfo", "r")
            val load = reader.readLine()
            reader.close()
            val tokens = load.split("\\s+".toRegex())
            tokens[1].toLong() / 1024
        } catch (e: Exception) {
            0L
        }
    }
}
